src/client/auth.rs

```diff
@@ -35,6 +35,11 @@ const MONGODB_AWS_STR: &str = "MONGODB-AWS";
 const MONGODB_X509_STR: &str = "MONGODB-X509";
 const PLAIN_STR: &str = "PLAIN";
 const MONGODB_OIDC_STR: &str = "MONGODB-OIDC";
```

Comment:
I feel like it's better defining all the constants like we did for the auth mechanisms